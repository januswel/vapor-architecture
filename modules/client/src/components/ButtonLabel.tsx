import * as React from 'react'
import { Image, View, StyleSheet, Text } from 'react-native'

const styles = StyleSheet.create({
  container: {
    width: 168,
    height: 168,
    margin: 16,
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  thumbnail: {
    width: 96,
    height: 96,
  },
  price: {
    fontSize: 16,
  },
})

const BACKGROUND_COLORS = {
  SOLD_OUT: {
    backgroundColor: 'red',
  },
  TOO_FEW_CHARGED_MONEY: {
    backgroundColor: 'gray',
  },
  AVAILABLE: {},
}

export interface Props {
  price: number
  imageUrl: string
  isSoldOut: boolean
  isTooFewChargedMoney: boolean
}

const computeBackgroundColor = (props: Props) => {
  if (props.isSoldOut) {
    return BACKGROUND_COLORS.SOLD_OUT
  }
  if (props.isTooFewChargedMoney) {
    return BACKGROUND_COLORS.TOO_FEW_CHARGED_MONEY
  }
  return BACKGROUND_COLORS.AVAILABLE
}

export default (props: Props) => {
  return (
    <View style={[styles.container, computeBackgroundColor(props)]}>
      <Image style={styles.thumbnail} source={{ uri: props.imageUrl }} resizeMode="contain" />
      <Text style={styles.price}>Â¥{props.price}</Text>
    </View>
  )
}
