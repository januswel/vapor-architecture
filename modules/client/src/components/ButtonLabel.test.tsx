import * as React from 'react'
import * as renderer from 'react-test-renderer'

import ButtonLabel from './ButtonLabel'

describe('', () => {
  it('renders button label for items sold out', () => {
    expect(
      renderer.create(
        <ButtonLabel
          price={120}
          imageUrl="http://example.com/assets/images/coffee.png"
          isSoldOut
          isTooFewChargedMoney
        />,
      ),
    ).toMatchSnapshot()

    expect(
      renderer.create(
        <ButtonLabel
          price={120}
          imageUrl="http://example.com/assets/images/coffee.png"
          isSoldOut
          isTooFewChargedMoney={false}
        />,
      ),
    ).toMatchSnapshot()
  })

  it('renders button label when charged money is too few to buy', () => {
    expect(
      renderer.create(
        <ButtonLabel
          price={120}
          imageUrl="http://example.com/assets/images/coffee.png"
          isSoldOut={false}
          isTooFewChargedMoney
        />,
      ),
    ).toMatchSnapshot()
  })

  it('renders button label for items which is available to buy', () => {
    expect(
      renderer.create(
        <ButtonLabel
          price={120}
          imageUrl="http://example.com/assets/images/coffee.png"
          isSoldOut={false}
          isTooFewChargedMoney={false}
        />,
      ),
    ).toMatchSnapshot()
  })
})
