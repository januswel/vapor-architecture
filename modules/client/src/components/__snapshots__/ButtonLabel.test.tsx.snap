// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button label for items sold out 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 168,
        "justifyContent": "space-around",
        "margin": 16,
        "width": 168,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "http://example.com/assets/images/coffee.png",
      }
    }
    style={
      Object {
        "height": 96,
        "width": 96,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 16,
      }
    }
  >
    짜
    120
  </Text>
</View>
`;

exports[`renders button label for items sold out 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 168,
        "justifyContent": "space-around",
        "margin": 16,
        "width": 168,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "http://example.com/assets/images/coffee.png",
      }
    }
    style={
      Object {
        "height": 96,
        "width": 96,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 16,
      }
    }
  >
    짜
    120
  </Text>
</View>
`;

exports[`renders button label for items which is available to buy 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 168,
        "justifyContent": "space-around",
        "margin": 16,
        "width": 168,
      },
      Object {},
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "http://example.com/assets/images/coffee.png",
      }
    }
    style={
      Object {
        "height": 96,
        "width": 96,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 16,
      }
    }
  >
    짜
    120
  </Text>
</View>
`;

exports[`renders button label when charged money is too few to buy 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 168,
        "justifyContent": "space-around",
        "margin": 16,
        "width": 168,
      },
      Object {
        "backgroundColor": "gray",
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "http://example.com/assets/images/coffee.png",
      }
    }
    style={
      Object {
        "height": 96,
        "width": 96,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 16,
      }
    }
  >
    짜
    120
  </Text>
</View>
`;
